docs/engineering/backend/readme.mdx

```diff
@@ -0,0 +1,138 @@
+        "value": "<app id>"
+    }
+    ```
+
+> **Note:** `identifier.value` above should correspond to `fhir_core_app_id` mentioned in the user management Keycloak section below.
+
+- Update the `sync_config.json` to remove all the non-patient data resources. These should be referenced from the Composition resource so they can be exempted from the Data filter. See [sync_config.json](https://github.com/opensrp/fhircore/blob/b7c24616d4224bd8d16c53b0c2a4f14a1075ce7c/android/quest/src/main/assets/configs/app/sync_config.json)
+
+## FHIR API and configuration resources
+
+1. Deploy the FHIR Store, e.g HAPI
+
+    - The steps here depend on what FHIR Store your are using. To deploy the HAPI FHIR Server using JPA, follow [these](https://github.com/hapifhir/hapi-fhir-jpaserver-starter) steps.
+
+2. `POST` the binary resources referenced in the `composition_config.json`
+
+    - `POST` all non-patient data resources used by the application. Examples of these resources are Questionnaires, StructureMaps, PlanDefinitions, Measures, Libraries, and any Lists referencing other resources.
+
+> **Note:**  When `POST`ing the Binary configuration files in the `composition_config.json` remember to first encode the payload to Base64 before submitting.
+
+> **Note:**  As described in the [FHIR Gateway](backend/info-gateway) section, the server should be in an internal network behind a DMZ and therefore not require authentication, which will be handled by the FHIR Information Gateway.
+
+## Deploy the FHIR Gateway
+
+1. Link to the [Docker image](https://hub.docker.com/r/onaio/fhir-gateway-plugin)
+1. The main documentation for deploying can be found in the [Github READ ME](https://github.com/onaio/fhir-gateway-plugin/blob/main/README.md)For configuration parameters, check out Read Me file for setting environment variables.
+1. For configuration parameters, check out Read Me file for setting environment variables.
+1. OpenSRP nuances: Provide/export  the System variable `ALLOWED_QUERIES_FILE` with value `"resources/hapi_page_url_allowed_queries.json"`[HAPI Page URL Allowed Queries](https://github.com/opensrp/fhir-gateway/blob/main/resources/hapi_page_url_allowed_queries.json)
+1. For each deployment the configuration entries for resources here should match the specific  `Composition` resource ID and `Binary` resources IDs
+
+- Provide System variable `SYNC_FILTER_IGNORE_RESOURCES_FILE` with value `"resources/hapi_sync_filter_ignored_queries.json"`[HAPI Sync Filter](https://github.com/opensrp/fhir-gateway/blob/main/resources/hapi_sync_filter_ignored_queries.json)
+
+    1. For the `List` entry add the specific ID of the List resource here. This enables us to use the List Resources outside of our configs use case
+
+- In the HAPI FHIR application.yaml disable validations by setting to `false*` [HAPI FHIR IPA Server](https://github.com/hapifhir/hapi-fhir-jpaserver-starter/blob/a06c0b9ce47a05f0719edeaba2a325ff5bef8cce/src/main/resources/application.yaml#L182)
```

Comment:
what do you set to false? all validations? I don't think we should encourage this, maybe we make it a note or a gotcha, but we shouldn't encourage it no matter what